Update installation docs (#10699)
dask/dask: docs/source/install.rst

=================

.. meta::
    :description: Dask Installation | You can easily install Dask with conda or pip

.. _Anaconda distribution: https://www.anaconda.com/download/

How to Install Dask
-------------------

You can install Dask with ``conda``, with ``pip``, or install from source.

.. tab-set::

   .. tab-item:: Conda

      If you use the `Anaconda distribution`_, Dask will be installed by default.

      You can also install or upgrade Dask using the
      `conda install <https://docs.conda.io/projects/conda/en/latest/commands/install.html>`_ command::

         conda install dask

      This installs Dask and **all** common dependencies, including pandas and NumPy.
      Dask packages are maintained both on the defaults channel and on
      `conda-forge <https://conda-forge.github.io/>`_.
      You can select the channel with the ``-c`` flag::

         conda install dask -c conda-forge

      Optionally, you can obtain a minimal Dask installation using the following command::

         conda install dask-core

>       This will install a minimal set of dependencies required to run Dask similar to (but not exactly the same as) ``python -m pip install dask`` below.

Reviewer: "Below" here felt a little off because of the tabs. Can probably just remove "below" altogether and not loose any meaning. 

```suggestion
      This will install a minimal set of dependencies required to run Dask similar to (but not exactly the same as) ``python -m pip install dask``.
```

      like pandas, Numpy, and others.

      You can also install only the Dask library and no optional dependencies::

         python -m pip install dask                # Install only core parts of dask

      Dask modules like ``dask.array``, ``dask.dataframe``, or
      ``dask.distributed`` won't work until you also install NumPy, pandas, or
      Tornado, respectively.  This is uncommon for users but more common for
      downstream library maintainers.

      We also maintain other dependency sets for different subsets of functionality::

         python -m pip install "dask[array]"       # Install requirements for dask array
         python -m pip install "dask[dataframe]"   # Install requirements for dask dataframe
         python -m pip install "dask[diagnostics]" # Install requirements for dask diagnostics
         python -m pip install "dask[distributed]" # Install requirements for distributed dask

      We have these options so that users of the lightweight core Dask scheduler
      aren't required to download the more exotic dependencies of the collections
      (Numpy, pandas, Tornado, etc.).

   .. tab-item:: Source

>       To install Dask from source, clone the repository from `github

Reviewer: Nit (also I realize this predates the changes here):

```suggestion
      To install Dask from source, clone the repository from `GitHub
```

      <https://github.com/dask/dask>`_::

         git clone https://github.com/dask/dask.git
         cd dask
         python -m pip install .

      You can also install all dependencies as well::

         python -m pip install ".[complete]"

>       You can view the list of all dependencies within the ``extras_require`` field
>       of ``setup.py``.

Reviewer: This predates the changes in this PR, but might as well fix since it's easy 

```suggestion
      You can view the list of all dependencies within the ``project.optional-dependencies`` field
      of ``pyproject.toml``.
```

      Or do a developer install by using the ``-e`` flag
      (see the :ref:`Install section <develop-install>` in the Development Guidelines)::

         python -m pip install -e .

> Distributed Deployment

Reviewer: Thanks for adding this section. It's a nice addition.

    cd dask
    py.test dask

> Installing Dask naively may not install all requirements by default (see the `Pip` section above).

Reviewer: Removing the `_` here also removes this as a link to the pip installation section above (currently rendered as "Pip" in italics)

Author: Yeah, we no longer have the section above.   I'll remove the accidental italics.

> Installing Dask naively may not install all requirements by default (see the `Pip` section above).

Author: ```suggestion
Installing Dask naively may not install all requirements by default (see the ``pip`` section above).
```
